### PR TITLE
feat(console,phrases): add third party app list page

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add third-party app list page

This PR includes the following updates:

1. Add new routes to display the third-party application tab
2. Update  the `navigate` method in `useTenantPathName` hook to support `navigate(-1)`
3. Add the new hook `useApplicationData` that wraps up the existing `first-party` application useSWR data fetching with newly added `third-party` application data fetching.
4. Keep a legacy copy of the original `first-party` application-only data fetching hooks for production use. Can be removed once the whole feature is ready.

## Testing

image

image

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/console/src/pages/Applications/hooks/use-application-data.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      buildUrl(applicationsEndpoint, {
+        page: String(thirdPartyApplicationPage),
+        page_size: String(pageSize),
+        isThirdParty: 'true',
+      }),
+    [thirdPartyApplicationPage]
+  );
+
+  const firstPartyApplicationsData = useSWR<[Application[], number], RequestError>(
+    firstPartyApplicationsFetchUrl
```
Comment (xiaoyijun) [inline packages/console/src/pages/Applications/hooks/use-application-data.ts:82]: Avoid redundant data fetching by
```suggestion
    !isThirdParty && firstPartyApplicationsFetchUrl
```

File: packages/console/src/pages/Applications/hooks/use-application-data.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  );
+
+  const thirdPartyApplicationsData = useSWR<[Application[], number], RequestError>(
+    thirdPartyApplicationsFetchUrl
```
Comment (xiaoyijun) [inline packages/console/src/pages/Applications/hooks/use-application-data.ts:86]: Ditto

File: packages/console/src/pages/Applications/hooks/use-legacy-application-data.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import { type Application } from '@logto/schemas';
+import { useCallback, useMemo } from 'react';
+import useSWR from 'swr';
+
+import { type RequestError } from '@/hooks/use-api';
+import useSearchParametersWatcher from '@/hooks/use-search-parameters-watcher';
+import { buildUrl } from '@/utils/url';
+
+const pageSize = 2;
```
Comment (xiaoyijun) [inline packages/console/src/pages/Applications/hooks/use-legacy-application-data.ts:9]:
```suggestion
const pageSize = 20;
```

**Review by xiaoyijun — CHANGES_REQUESTED** [review]

File: packages/console/src/pages/Applications/hooks/use-legacy-application-data.ts (hunk 3 above)
Comment (simeng-li) [reply packages/console/src/pages/Applications/hooks/use-legacy-application-data.ts:9]: should be defaultPageSize, 2 is to testing use.

**Review by xiaoyijun — APPROVED** [review]
LGTM

**Pull request merged**